src/Spanner/Session/CacheSessionPool.php

```diff
@@ -777,4 +764,63 @@ private function validateConfig()
             );
         }
     }
+
+    /**
+     * Delete the provided sessions.
+     *
+     * @param array $sessions
+     */
+    private function deleteSessions(array $sessions)
+    {
+        $calls = [];
+
+        foreach ($sessions as $session) {
+            $calls[] = $this->database->connection()
+                ->deleteSessionAsync([
+                    'name' => $session['name'],
+                    'database' => $this->database->name()
+                ]);
+        }
+
+        foreach ($calls as $call) {
+            $call->wait();
```

Comment:
/cc @vkedia

This is our last blocker before we push Spanner to beta. If you have a moment, it would be very much appreciated to get your sign off on the approach before we merge.

Mike and I have been discussing options offline a bit and would like to propose the following:

We provide a configuration option to the pool (`shouldDeleteOnShutdown`) which will determine whether or not to register a shutdown function which will call `wait`<sup>1</sup> to block on the delete RPCs completing.  The delete RPCs will be happening concurrently while the user runs their desired RPC (executeSql, for instance) - so the delay from calling `wait` should be very minimal. In the case one of the delete RPCs hangs longer than expected, `wait` would end up blocking for an undetermined amount of time. To help mitigate the effects of this, we can set a very short timeout on the delete RPCs.

Let us know what you think!

1. It appears calling `wait` may be the best way to guarantee a session will be deleted. If the PHP script dies before the delete RPC finishes after calling `start` it will give up on attempting to complete the request. For example, if the delete RPC takes 100ms but executeSql takes 50ms and the script ends after executeSql completes, the delete RPC will not finish. If preferred, we can omit wait and just allow any sessions whose delete RPC fails to just expire on their own. The benefit of omitting wait is that we could avoid any added latency if the delete RPCs do end up hanging and the timeout is not sufficient enough.